src/Lykke.Bil2.RabbitMq/Subscription/Core/BackgroundService.cs

```diff
@@ -0,0 +1,76 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Nito.AsyncEx;
+
+namespace Lykke.Bil2.RabbitMq.Subscription.Core
+{
+    internal abstract class BackgroundService : IBackgroundService
+    {
+        private readonly CancellationTokenSource _stoppingCts;
+        
+        private Task _executingTask;
+
+        
+        protected BackgroundService()
+        {
+            _stoppingCts = new CancellationTokenSource();
+        }
+
+        protected abstract Task ExecuteAsync(
+            CancellationToken stoppingToken);
+
+        public virtual void Start()
+        {
+            _executingTask = ExecuteAsync(_stoppingCts.Token);
```

Comment:
Wrap execution in this base class by try/catch and log unhandled exceptions. Additionally would be good to move outer ```while (!stoppingToken.IsCancellationRequested)``` cycle to the base class too and catch unhandled exceptions inside the loop.

Reply:
Added